iceoryx_systemtest/src/iox_publisher_systemtest.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "topic_data.hpp"
+
+#include "iceoryx_posh/popo/untyped_publisher.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+
+#include <iostream>
+
+bool killswitch = false;
+
+static void sigHandler(int f_sig [[gnu::unused]])
+{
+    // caught SIGINT or SIGTERM, now exit gracefully
+    killswitch = true;
+}
+
+int main()
+{
+    // Register sigHandler for SIGINT
+    signal(SIGINT, sigHandler);
```

Comment:
Please use sigaction instead of signal since the latter has some issues
https://stackoverflow.com/a/232711

I'm fully aware that we don't fully do this in our code base, but we should strive to replace it with sigaction. Creating a nice abstraction around sigaction would be a good first step.